src/org/zaproxy/zap/extension/fuzz/payloads/generator/RegexPayloadGenerator.java

```diff
@@ -71,26 +78,198 @@ public long getNumberOfPayloads() {
     }
 
     @Override
     public RegexPayloadGenerator copy() {
-        return new RegexPayloadGenerator(regex, maxPayloads, maxPayloadLength, numberOfPayloads);
+        return this;
+    }
+
+    /**
+     * Tells whether or not the given {@code regex} is syntactically valid for the payload generator.
+     *
+     * @param regex the regular expression that will be validated
+     * @return {@code true} if the {@code regex} is syntactically valid, {@code false} otherwise
+     * @see #isValid(String)
+     */
+    public static boolean isSyntacticallyValid(String regex) {
+        if (regex == null) {
+            throw new IllegalArgumentException("Parameter regex must not be null.");
+        }
+        return Generex.isValidPattern(regex);
     }
 
-    public static int calculateNumberOfPayloads(String regex, int maxPayloads, int maxPayloadLength) {
+    /**
+     * Tells whether or not the given {@code regex} is valid for the payload generator.
+     * <p>
+     * A regular expression might be syntactically valid but still invalid for the payload generator if it takes too much time
+     * to be processed.
+     *
+     * @param regex the regular expression that will be validated
+     * @return {@code true} if the {@code regex} is valid, {@code false} otherwise
+     * @throws IllegalArgumentException if the given {@code regex} is {@code null}
+     * @see #isSyntacticallyValid(String)
+     */
+    public static boolean isValid(final String regex) {
+        if (!isSyntacticallyValid(regex)) {
+            return false;
+        }
+        return TimeOutRunner.run(new Runnable() {
+
+            @Override
+            public void run() {
+                @SuppressWarnings("unused")
+                Generex generator = new Generex(regex);
+            }
+        }, VALID_REGEX_MAX_SECONDS, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Tells whether or not the given {@code regex} is infinite, that is, generates an infinite number of payloads, taking into
+     * account the given {@code limit}.
+     *
+     * @param regex the regular expression that will be validated
+     * @param limit if positive, the maximum number of payloads that are allowed, otherwise, negative or zero, for no limit
+     * @return {@code true} if the {@code regex} is infinite, {@code false} otherwise
+     * @throws IllegalArgumentException if the given {@code regex} is {@code null} or not valid
+     * @see #isValid(String)
+     */
+    public static boolean isInfinite(String regex, int limit) {
+        validateValid(regex);
+        return isInfiniteImpl(new Generex(regex), limit);
+    }
+
+    private static void validateValid(String regex) {
+        if (regex == null) {
+            throw new IllegalArgumentException("The provided regular expression must not be null.");
+        }
+        if (!isValid(regex)) {
+            throw new IllegalArgumentException("The provided regular expression must be valid: " + regex);
```

Comment:
Those exception messages are not expected to be shown in the GUI or directly to the user. That's just for help/inform us, in case something goes wrong.